file/example_test.go

```diff
@@ -40,3 +40,53 @@ func Example() {
 	// 1
 	// /topic
 }
+
+func ExampleSubscriber() {
+	dir, err := ioutil.TempDir("", "bps-example")
+	if err != nil {
+		panic(err.Error())
+	}
+	defer os.RemoveAll(dir)
+
+	ctx := context.TODO()
+
+	// produce some messages (seed topic):
+	SeedTopic(ctx, dir, "topic")
+
+	sub, err := bps.NewSubscriber(ctx, "file://"+dir)
+	if err != nil {
+		panic(err.Error())
+	}
+	defer sub.Close()
+
+	handler := bps.HandlerFunc(func(msg bps.SubMessage) error {
+		fmt.Printf("%s\n", msg.Data())
+		return nil // or bps.Done to stop/unsubscribe
+	})
+
+	// blocks till all the messages are consumed or error occurs:
+	err = sub.Subscribe(ctx, "topic", handler)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// Output:
+	// {"data":"bWVzc2FnZQ=="}
```

Comment:
that is strange given that you print msg.Data as a string? this isn't quite right. your "Data" is "message"